source/speech/speech.py

```diff
@@ -1169,9 +1170,28 @@ def speak(  # noqa: C901
 			)
 			if not inCharacterMode:
 				speechSequence[index] += CHUNK_SEPARATOR
+				if curLanguage not in (defaultLanguage, SpeechState.lastReportedLanguage):
+					speechSequence.insert(0, LangChangeCommand(defaultLanguage))
+					speechSequence.insert(1, languageHandler.getLanguageDescription(curLanguage))
+					SpeechState.lastReportedLanguage = curLanguage
+					if not languageIsSupported(curLanguage):
+						log.warn(f"{curLanguage} not supported in {getSynth().name}")
 	_manager.speak(speechSequence, priority)
 
 
+def languageIsSupported(language: str | None) -> bool:
+	"""Determines if the specified language is supported.
+	:param language: A language code or None.
+	:return: True if the language is supported, False otherwise.
+	"""
+	if language is None:
+		return True
+	for lang in getSynth().availableLanguages:
+		if language == lang or language == languageHandler.normalizeLanguage(lang).split("_")[0]:
```

Comment:
Should this piece of code depend on the "Automatic dialect switching (when supported)" option?

Reply:
I think it's better that this doesn't deppend on the automatic dialect switching option, since we can imagine a synthesizer with languages just in the form of dialects. Anyway I'm open to change my opinion.